### PR TITLE
#3804 Show file path on error

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: #3804

### What are the changes and their implications?

- [x] wrap the logic of the steps inside the ```getAllFiles``` function with a ```try...catch``` block to get the file name during errors

### CODE REVIEW

File: packages/codemod/src/upgrade-legacy.ts [hunk 1]
```diff
@@ -284,37 +284,42 @@ const upgradeLegacy = async () => {
...
-                const index = parsedProgram.value.program.body.indexOf(e)
-                parsedProgram.value.program.body.splice(index, 1)
               }
             }
-          }
-        })
-        fs.writeFileSync(path.resolve(appDir, filename), program.toSource())
+          })
+          fs.writeFileSync(path.resolve(appDir, filename), program.toSource())
+        } catch (e: any) {
```
Comment (dillonraphael) [inline packages/codemod/src/upgrade-legacy.ts:319]: We're being very strict with using any or assertions. You can see how to handle errors here https://github.com/blitz-js/blitz/blob/39fd15d4b5a522740197d1311db16884bd7dfc91/packages/codemod/src/upgrade-legacy.ts#L1301

**Review by dillonraphael — CHANGES_REQUESTED** [review]

File: packages/codemod/src/upgrade-legacy.ts (hunk 1 above)
Comment (siddhsuresh) [reply packages/codemod/src/upgrade-legacy.ts]: hey, @dillonraphael I have removed all the manual assertions to ```any``` for errors that can be caught in a file. The inner ```try...catch``` will just pass on the error to the outer catch block like you mentioned that should do the check. I was not sure whether to duplicate the logic in all places. If this is what is expected, I shall do that.

Comment (dillonraphael) [reply packages/codemod/src/upgrade-legacy.ts]: Yup passing it to the outer catch block should be good @siddhsuresh. But seems the CI is failing.

Comment (siddhsuresh) [reply packages/codemod/src/upgrade-legacy.ts]: @dillonraphael The CI is now passing.

**Review by dillonraphael — APPROVED** [review]

**Pull request merged**